test/src/tests-vault.ts

```diff
@@ -1,329 +1,168 @@
 import { Batch } from '@mparticle/event-models';
 import { expect } from 'chai';
+import { Dictionary } from '../../src/utils';
 import Vault from '../../src/vault';
 
 describe('Vault', () => {
     afterEach(() => {
         window.localStorage.clear();
     });
 
-    describe('#storeItems', () => {
-        it('should store an individual item as a hashmap', () => {
-            const batch1: Partial<Batch> = {
-                mpid: 'bar',
-                source_request_id: 'item-123',
-            };
+    describe('#store', () => {
+        it('should store an object', () => {
+            const storageKey = 'test-key-store-object';
 
-            const batch2: Partial<Batch> = {
-                mpid: 'baz',
-                source_request_id: 'item-456',
+            const dict: Dictionary<Dictionary<string>> = {
+                foo: { foo: 'bar', buzz: 'bazz' },
+                pinky: { narf: 'poit' },
             };
 
-            const vault = new Vault<Partial<Batch>>(
-                'test-key-store-batches',
-                'source_request_id',
-                { offlineStorageEnabled: true }
+            const vault = new Vault<Dictionary<Dictionary<string>>>(storageKey);
+
+            vault.store(dict);
+
+            expect(vault.contents).to.equal(dict);
+            expect(window.localStorage.getItem(storageKey)).to.equal(
+                JSON.stringify(dict)
             );
+        });
+
+        it('should store an array', () => {
+            const storageKey = 'test-key-store-array';
 
-            vault.storeItem(batch1);
+            const array: Dictionary<string>[] = [
+                { foo: 'bar', buzz: 'bazz' },
+                { narf: 'poit' },
+            ];
 
-            expect(vault.contents).to.eql({ 'item-123': batch1 });
+            const vault = new Vault<Dictionary<string>[]>(storageKey);
 
-            vault.storeItem(batch2);
+            vault.store(array);
 
-            expect(vault.contents).to.eql({
-                'item-123': batch1,
-                'item-456': batch2,
-            });
+            expect(vault.contents).to.equal(array);
+            expect(window.localStorage.getItem(storageKey)).to.equal(
+                JSON.stringify(array)
+            );
         });
-    });
 
-    describe('#storeItems', () => {
-        it('should store an array of batches as a hashmap', () => {
-            const batch1 = {
-                foo: 'bar',
+        it('should store batches in the order they were added', () => {
+            const storageKey = 'test-batch-save-order';
+
+            const batch1: Partial<Batch> = {
+                mpid: 'bar',
```

Comment:
Very minor nit.  There should be some sort of order to each mpid and source_request_id for each batch. I know we aren't necessarily referencing those items.  But I might do something like `source_request_id_1/2/3/4/5`, and `mpid1/2/3/4/5` rather than them being totally random. It creates a bit more order and is easier to read.

Reply:
I think my intention was to show that the order wasn't being alphabetized, which is an issue with JS objects.

Reply:
Oh - let me rephrase - it has more to do with readability so we would know which mpid and which source request id was part of which batch if we ever needed to test those.  I know we're not testing them specifically in this test, but we should adhere to that practice to make it easier to grok that `mpid2` is with `batch2`, `mpid1` is `batch1`, rather than how it is now with mpid of `bar` with `batch1`, mpid `baz` with `batch2`.